### PR TITLE
[AN-5520] fixes infinite loop in the input text (on top of "rewrite cursor" PR)

### PR DESCRIPTION
**Pull request by makingthematrix**
#### APK
[Download build #9312](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9312/artifact/build/artifact/wire-dev-PR1067-9312.apk)
[Download build #9320](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9320/artifact/build/artifact/wire-dev-PR1067-9320.apk)
[Download build #9322](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9322/artifact/build/artifact/wire-dev-PR1067-9322.apk)
[Download build #9323](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9323/artifact/build/artifact/wire-dev-PR1067-9323.apk)
[Download build #9324](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9324/artifact/build/artifact/wire-dev-PR1067-9324.apk)
[Download build #9325](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9325/artifact/build/artifact/wire-dev-PR1067-9325.apk)

### CODE REVIEW

**Comment by deanrobertcook** [comment]
@makingthematrix - that `Merge branch rel/2.39.x` commit seems a little off...

**Comment by deanrobertcook** [comment]
yup, looks good - let's maybe also fix the cursor colour and we should be good to try this again

File: app/src/main/scala/com/waz/zclient/cursor/CursorView.scala [hunk 1]
```diff
@@ -136,21 +136,28 @@ class CursorView(val context: Context, val attrs: AttributeSet, val defStyleAttr
   secondaryToolbar.cursorItems ! SecondaryCursorItems
 
   cursorEditText.addTextChangedListener(new TextWatcher() {
+    private var text = ""
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/cursor/CursorView.scala]: Also - remember to be a little more defensive with stuff from the Android system - maybe on some devices it's possible that `charSequence` can be null - so best to use an option here

File: app/src/main/java/com/waz/zclient/pages/main/conversation/ConversationFragment.java [hunk 2]
```diff
@@ -444,51 +433,49 @@ public View onCreateView(LayoutInflater inflater, ViewGroup viewGroup, Bundle sa
...
+                    }
+                    return false;
+                }
+            });
+            toolbar.setNavigationOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    if (cursorView == null) {
+                        return;
+                    }
```
Comment (deanrobertcook) [inline app/src/main/java/com/waz/zclient/pages/main/conversation/ConversationFragment.java]: The indenting is off here

File: app/src/main/java/com/waz/zclient/pages/main/conversation/ConversationFragment.java [hunk 3]
```diff
@@ -444,51 +433,49 @@ public View onCreateView(LayoutInflater inflater, ViewGroup viewGroup, Bundle sa
...
-            cursorLayout.closeEditMessage(false);
-            getActivity().onBackPressed();
-            KeyboardUtils.closeKeyboardIfShown(getActivity());
-            }
-        });
+                    cursorView.closeEditMessage(false);
+                    getActivity().onBackPressed();
+                    KeyboardUtils.closeKeyboardIfShown(getActivity());
+                }
+            });
```
Comment (deanrobertcook) [inline app/src/main/java/com/waz/zclient/pages/main/conversation/ConversationFragment.java]: and here

**Comment by deanrobertcook** [comment]
Looks like we can also get rid of the `api.InputStateIndicator`

**Comment by someoneEsle** [comment]
1. type something in conversation with A, don't send
2. change theme (from light to dark, or dark to light)
3. open conversation with A
4. what you typed isn't there anymore

**Comment by someoneEsle** [comment]
1. open conversation with A, set timed mode, type something without sending
2. change your accent color
3. go back to conversation with A
4. send arrow does not have the right accent color

**Comment by someoneEsle** [comment]
All looks good

**Pull request merged**